corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -417,7 +429,6 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         submitAction: function (e) {
             e.preventDefault();
             FormplayerFrontend.trigger('clearNotifications', errorHTML, true);
-
             var invalidFields = [];
             this.children.each(function (childView) {
                 if (!childView.isValid()) {
```

Comment:
I had some time today to look at this. At the moment the error message is not showing up on the first submit, but is showing after the second. If I'm understanding right that's because currently the `error message` is being set in `isValid` while the `error` on the model is being set by formplayer after submit. Some communication with formplayer needs to happen somewhere else, possibly is the call to `_isValid`. I'm leaving this note as reminder for tomorrow, but also happy to have my logic checked and take suggestions. @orangejenny This feels very similar to what you were saying about the mechanics of validating on change and I need to look at that again. @shubham1g5 @esoergel